Add ready-player

xenodium opened the pull request:
### Brief summary of what the package does

A major mode to open media (audio/video) files in an Emacs buffer.

![Screengrab](https://github.com/melpa/melpa/assets/8107219/adeb124b-d848-41ab-88a0-41b47785e023)

### Direct link to the package repository

https://github.com/xenodium/ready-player

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks for this.  Quick first pass --

`ready-player.el` with byte-compile using Emacs 29.4:
```
ready-player.el:527:28: Warning: Unused lexical variable `recentf-exclude'
In end of data:
ready-player.el:473:5: Warning: the function `text-property-search-backward' is not known to be defined.
```

`ready-player.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- ready-player.el#L281: It's safer to sharp-quote function names; use `#'`
- ready-player.el#L716: It's safer to sharp-quote function names; use `#'`
- ready-player.el#L819: It's safer to sharp-quote function names; use `#'`
- ready-player.el#L253: It's safer to sharp-quote function names; use `#'`
- ready-player.el#L232: You may want to prefer `(derived-mode-p 'xyz)`
- ready-player.el#L659: You may want to prefer `(derived-mode-p 'xyz)`
- ready-player.el#L694: You may want to prefer `(derived-mode-p 'xyz)`
```

xenodium commented:
Thank you @riscy!

```
ready-player.el:527:28: Warning: Unused lexical variable `recentf-exclude'
In end of data:
ready-player.el:473:5: Warning: the function `text-property-search-backward' is not known to be defined.
```

Oooh, I had just fixed prolly just around the time you were reviewing.

I've just ran melpazoid on latest:

```
MELPA_PR_URL=https://github.com/melpa/melpa/pull/9096 make
python3 melpazoid/melpazoid.py
<!-- Cloning https://github.com/xenodium/ready-player.git  -->
<!-- Building container for ready-player... 🐳 -->
sha256:47b960a22bfd8877044e25baba138b015b9b1fa9211b72f32eaf5c6dbf2c26df

<!-- Running melpazoid tests... -->
sha256:47b960a22bfd8877044e25baba138b015b9b1fa9211b72f32eaf5c6dbf2c26df

`ready-player.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- ready-player.el#L258: Prefer `add-to-list` to add to auto-mode-alist
```

Package and license:
- GNU General Public License v3.0 (via API)
- GPL* -- ready-player.el: Open media files in ready-player-mode major mode


<!-- Footnotes:
- (ready-player
    :fetcher github
    :repo "xenodium/ready-player")
- Created: 2024-07-03T14:35:03Z
- Updated: 2024-07-14T16:42:21Z
- Watched: 73
-->
```

Looks like the remaining is a false positive?

riscy commented:
> Looks like the remaining is a false positive?

Yes, I stripped that out of my feedback last time because it's a false positive.  Last time I did include these --

- ready-player.el#L278: You may want to prefer `(derived-mode-p 'xyz)`
- ready-player.el#L775: You may want to prefer `(derived-mode-p 'xyz)`
- ready-player.el#L811: You may want to prefer `(derived-mode-p 'xyz)`

which is extremely minor but may want to be something you consider (or not).

Another minor thing, I'm noticing in this documentation you don't have to escape the `=`, and you may be missing an `or` on the end.  
```
Last button focused (\=next\= or \=previous\= \=play-stop\=).
```

And one other thing I'd warn against is this --
```elisp
                    ;; Execute after current run loop to allow `find-file' completion.
                    (run-at-time 0 nil ...)
```
Unless I'm missing something, I feel like this may be relying on undocumented assumptions about how `run-at-time` is currently implemented/behaves.  E.g. some future optimization might make this code run *before* the current loop is complete.  I think a nonzero value might be safer, even though I think you'll still have some kind of race going on.

No blockers though - merging now.  And thank you!

xenodium commented:
Thank you for the feedback and help @riscy! Much appreciated.

I've addressed all comments at https://github.com/xenodium/ready-player/commit/4ef131a9f55b4c79ce5deb568e7c815fe4b08c1b

```
    ready-player.el#L278: You may want to prefer (derived-mode-p 'xyz)
    ready-player.el#L775: You may want to prefer (derived-mode-p 'xyz)
    ready-player.el#L811: You may want to prefer (derived-mode-p 'xyz)
```

While I've now fixed these, I never managed to get them flagged. I ran:

```
MELPA_PR_URL=https://github.com/melpa/melpa/pull/9096 make
```
